src/Symfony/Component/Translation/LoggingTranslator.php

```diff
@@ -82,7 +82,9 @@ public function transChoice($id, $number, array $parameters = [], $domain = null
      */
     public function setLocale($locale)
     {
+        $prev = $this->translator->getLocale();
         $this->translator->setLocale($locale);
+        $this->logger->debug('Translator has changed the locale.', ['previous' => $prev, 'new' => $locale]);
```

Comment:
+1 it can be usefull in dev

previous/next or old/new, here you use one of both

Reply:
It'd suggest putting the locale in the message, e.g.:
`The locale of the translator has changed from "%s" to "%s".`